tests/strings/test_z_algorithm.py

```diff
@@ -25,14 +22,26 @@ def generate_words(char_set, length):
 
 def test_z_algorithm__corner_cases():
     for c in range(1, 4):
-        for length in range(1, 11):
-            for word in generate_words(ascii_lowercase[:c], length):
-                assert brute_force(word) == z_function(word)
+        for n in range(1, 11):
+            for text in generate_words(ascii_lowercase[:c], n):
+                assert z_function(text) == [get_lcp(text, i) for i in range(n)]
 
 
 def test_z_algorithm__large_cases():
-    for c in range(1, 10):
-        length = 10**5
-        text = ''.join(choice(ascii_lowercase[:c]) for _ in range(length))
+    for c in range(1, 4):
+        n = 10**5
+        text = ''.join(choice(ascii_lowercase[:c]) for _ in range(n))
+
         z_array = z_function(text)
-        assert z_array[:100] == brute_force(text, 100)
+        for _ in range(100):
+            i = randint(0, n - 1)
```

Comment:
just a note, there exists randrange, which is arguably nicer here since then you can just do `i = randrange(n)`.

Reply:
Sure, will update.